### ISSUE TITLE
Improve support for previews in deployed environments

### ISSUE THREAD

**Issue by aduth** [allmarkedup/lookbook#73]
I would love to be able to use Lookbook in a deployed environment to be able to share preview links to other designers and developers on our team. However, we encountered some challenges when trying to do this, since it appears that Lookbook makes some assumptions that it is running in a local environment.

The two specific issues we encountered are:

1. Parsing at runtime
   - Problem: In our deployed environment, file permissions are such that YARD parsing fails due to an attempt to [create a file write lock](https://github.com/allmarkedup/lookbook/blob/0a6d894946f0b1ba3e25f0c30fee42690d381b16/lib/lookbook/parser.rb#L14) at runtime
   - Possible solution: Allow for pre-parse to occur, possibly as part of assets precompilation
   - Relevant source: https://github.com/allmarkedup/lookbook/blob/0a6d894946f0b1ba3e25f0c30fee42690d381b16/lib/lookbook/engine.rb#L120
2. File watching
   - Problem: Lookbook listens for changes to files, despite the fact that files would not be expected to change in a deployed environment
   - Possible solutions: 
      - New option to opt-out of file watching
      - Or: Let [`options.listen_paths`](https://github.com/allmarkedup/lookbook/blob/0a6d894946f0b1ba3e25f0c30fee42690d381b16/lib/lookbook/engine.rb#L52) take precedence over [default paths](https://github.com/allmarkedup/lookbook/blob/0a6d894946f0b1ba3e25f0c30fee42690d381b16/lib/lookbook/engine.rb#L53-L54)
      - Or: Disable file watching based on Rails environment (e.g. `Rails.env.production?`)
   - Relevant source: https://github.com/allmarkedup/lookbook/blob/0a6d894946f0b1ba3e25f0c30fee42690d381b16/lib/lookbook/engine.rb#L85

**Comment by allmarkedup** [comment]
Hey @aduth - thanks for this. Lookbook was really initially intended to be a development environment for ViewComponent components so up until now I've not put much thought into the production environment side of things. However I know that you are not alone in wanting to do this, and with the addition of 'pages' functionality in the latest release it makes total sense to support it.

I really appreciate you highlighting the issues you ran into when trying to deploy it - very useful, thanks. I'm working to try and resolve some performance issues and bugs in the latest beta release but once I've addressed those I'll look to make support for production deployments a priority.

As soon as I've got a solution for the issues you are running into I'll update you here and hopefully unblock things for you.

**Comment by allmarkedup** [comment]
Hi again @aduth. Off the back of your issue I've done some work to hopefully make Lookbook more compatible with being deployed into production environments.

I've just release a beta version that you can try (0.8.0.beta.0) with some documentation here about the changes: https://github.com/allmarkedup/lookbook/tree/v0.8.0-beta.0#deployment

If you have a chance to try it out I'd welcome any further feedback or comments you might have. Many thanks.

**Comment by aduth** [comment]
Thanks @allmarkedup for the quick turnaround! Based on a quick glance of the documentation, I think it'll work great. I'll plan to give the beta version a try this upcoming week, and I'll let you know how it goes.

**Comment by allmarkedup** [comment]
Ok great, sounds good. Any problems just let me know!

**Comment by aduth** [comment]
Initial testing is promising! At least, it no longer crashes when it did previously. 😅  I'm still sorting out some conflicts with our build process for making the pre-parsed YARDoc available. The other main issue for us is going to be content-security policy, since apparently [Alpine.js relies on some unsafe evaluation by default](https://alpinejs.dev/advanced/csp). Since we probably wouldn't host this in our user-facing production environment, it could be reasonable to assume more lax rules, at least for these routes. In any case, it could be a conflict to expect for others who might have a CSP. I'm also not familiar enough with Alpine.js to know if their "CSP alternate build" is an easy enough swap.

**Comment by allmarkedup** [comment]
Hey @aduth I just wanted to check in and see if you go this working in the end? If it's working for you now then I'll go ahead and close this issue down.

Just FYI, I looked into the Alpine CSP build and it's not something that will be an easy port I'm afraid. The CSP-compatible build isn't actually even properly released as a package yet either (despite what the docs hint at) so I don't think it's a good place to put my efforts at the moment. Sorry!

**Comment by aduth** [comment]
Hi @allmarkedup , I haven't been able to return to this, but given that the issues I was seeing were specific to our environment, and that the feature has now shipped (🎉 ), happy to close this issue and open a new issue if I encounter any problems once I can revisit.

Thanks again for considering this!

**Comment by allmarkedup** [comment]
That's great @aduth, and please do let me know if there is anything else that could make deployments (or anything else!) easier for you in the future 👍
